Enable posts to have a list of authors

charlieegan3 opened the pull request:
Sometimes posts have more than one author, it'd be nice to represent
this on post pages.

This reads from `author` or `authors` from the front matter to build a
list of authors. The list of authors is used to draw the page (list of
names at the top, list of 'profiles' at the bottom).

charlieegan3 committed:
Enable posts to have a list of authors

Sometimes posts have more than one author, it'd be nice to represent
this on post pages.

This reads from `author` or `authors` from the front matter to build a
list of authors. The list of authors is used to draw the page (list of
names at the top, list of 'profiles' at the bottom).

charlieegan3 committed:
Handle guest authors and feed indexes

tomsabin reviewed (dismissed):
Two things:
1. Could we change the base of this branch to staging?
2. We are missing the authors on the index page when there are multiple:
   ![localhost-4567-blog iphone 4](https://cloud.githubusercontent.com/assets/885223/19886824/0181b5cc-a01d-11e6-86be-b81e02fdbbff.png)

with author:
![localhost-4567-blog iphone 4 1](https://cloud.githubusercontent.com/assets/885223/19886846/1795eb76-a01d-11e6-84c3-f2b1dfa870a4.png)

I'm not sure if that was intentional but we do then have an empty div

h-lame commented:
We've got the logic for finding the author spread across 4 templates now (we already did I think) but now it's more complex (author vs authors) so I think it's time to write a helper or something to extract the authors correctly.

charlieegan3 commented:
Thanks @h-lame and @tomsabin; I could pick this up later on but I've asked @cih to pick it up for the time being so this doesn't holdup their blog post.

cih committed:
Add article info containing logic for article authors

This change moves the logic for determining the authors of an article
to a single location.

cih committed:
Add article info helper and update blog layout

This updates the blog layout page to use the article info to get the
authors and current authors. This will now handle the case when someone
who has coauthored an article and since been removed from the people.yml

cih committed:
Extract find_person method

cih committed:
Update blog feed to use article info

cih committed:
Add author_names method and update blog tiles

This change updates the blog tiles to include multiple authors

h-lame commented on lib/former_employee.rb:
```diff
@@ -0,0 +1,27 @@
+class FormerEmployee
```
Nice to see a null object here.  One problem is that not all authors who don't appear in the `data.people` are ex employees - we've had some guest bloggers before and maybe even clients?  So perhaps `NonEmployee` would be more correct (albeit not as snappy, better names are possible I'm sure).

cih replied:
Ah yes I hadn't realised that, thanks I will update.

h-lame commented on source/blog/2016-10-28-next-london-2016-google-cloud-platform-s-user-conference.md:
```diff
@@ -2,9 +2,9 @@
 title: NEXT London 2016 - Google Cloud Platform's User Conference
-authors: 
+author: Charlie Egan
+coauthors:
```
Why the split from author into author + coauthors ?  It implies a hierarchy of authorship that we might not intend (how to decide who is the author and who is the coauthor, especially meaningless as we don't distinguish in the page rendering).  I'd be in favour of saying we don't have author anymore, just authors (and do a grep to fix all the old frontmatter).

cih replied:
The reason I changed it from the initial author and authors is that I didn't want to have them being one or the other. But agree that if we remove the author and just go with authors consistently that will be good.

cih committed:
Rename former employee to non employee

Suggested in review as we also have guest posts from people who
are not former employees.

cih committed:
Update article info to use authors array

Instead of using author and coauthors we will use an authors array
and update existing posts to this format.

cih committed:
Update all old blog posts to use authors

Recent changes updated author to authors to allow us to support
multiple authors.

cih committed:
Update article weighting to use the authors array

This change updates the logic in the related blog articles weighting
calculation to use array intersection.

cih committed:
Update blog author grid and article

This updates the templates to check for the author name in the
authors array.

cih committed:
Update blog related tiles to show author names

cih committed:
Update homepage blog tiles to use author names

cih committed:
Remove space between author name and comma
